### PR TITLE
feat: move runner scaling to the provisioner

### LINKED ISSUE #3483: Refactor k8s provisioning to be a provisioner

### PR DESCRIPTION
**Pull request by stuartwdouglas**
fixes #3483

### CODE REVIEW

File: backend/provisioner/service.go [hunk 1]
```diff
@@ -149,6 +150,8 @@ func replaceOutputs(to []*provproto.Resource, from []*provproto.Resource) error
 				}
 				r.Subscription.Output = subscriptionFrom.Subscription.Output
 			}
+		case *provproto.Resource_Runner:
+			// Ignore
```
Comment (jvmakine) [inline backend/provisioner/service.go:154]: Why ignoring here? Should we not just set the outpus?

File: backend/provisioner/registry.go [hunk 2]
```diff
@@ -273,9 +276,18 @@ func ExtractResources(msg *ftlv1.CreateDeploymentRequest) (*ResourceGraph, error
 			to:   dep.ResourceId,
 		})
 	}
-
+	runnerResource := &provisioner.Resource{
+		ResourceId: module.GetName() + "-runner",
```
Comment (jvmakine) [inline backend/provisioner/registry.go:280]: Should we use deployment id here instead? I think that would fit better to the given model, given that each deployment has a separate runner as a resource.

**Review by jvmakine — APPROVED** [review]

File: backend/provisioner/registry.go (hunk 2 above)
Comment (stuartwdouglas) [reply backend/provisioner/registry.go:280]: Good point

Comment (stuartwdouglas) [reply backend/provisioner/registry.go:280]: Actually we don't really seem to have the deployment name here?

Comment (jvmakine) [reply backend/provisioner/registry.go:280]: Hmm, yeah. I think it currently created after the provisioning, which is annoying. I suppose that is fine for now, though it breaks the model a bit.

**Pull request merged**